### ISSUE TITLE
mtmd: gemma4uv projector produces NaN logits -> sampler assertion failure with use_gpu=true

### ISSUE THREAD

**Issue by lienzzzz** [ggml-org/llama.cpp#24251]
## Description

When using Gemma 4 mmproj (`gemma4uv` projector type) with GPU backend enabled, the model produces NaN/invalid logits after processing image input, causing an assertion failure in the sampler.

## Environment

- **llama.cpp commit:** `6b80c74f285390368b3c99c5e750f19e9b096e98` (master, 2026-06-07)
- **GPU:** NVIDIA RTX 5070 Ti (Blackwell, compute capability 12.0)
- **CUDA:** 13.0
- **OS:** Windows 10

## Model

- GGUF: `gemma-4-12B-it-abliterated-uncensored.Q6_K.gguf`
- mmproj: `gemma-4-12B-it-abliterated-uncensored.mmproj-f16.gguf` (projector type: `gemma4uv`, 11 tensors, has vision + audio encoder)

## Steps to Reproduce

```python
from llama_cpp import Llama
from llama_cpp.llama_chat_format import (
    Gemma4ChatHandler, Jinja2ChatFormatter, register_chat_format
)

CHAT_FORMAT_NAME = "gemma4_custom"
template = Gemma4ChatHandler.CHAT_FORMAT
formatter = Jinja2ChatFormatter(
    template=template,
    eos_token=Gemma4ChatHandler.GEMMA4_EOS_TOKEN,
    bos_token=Gemma4ChatHandler.GEMMA4_BOS_TOKEN,
)

@register_chat_format(CHAT_FORMAT_NAME)
def _fmt(*args, **kwargs):
    return formatter(*args, **kwargs)

chat_handler = Gemma4ChatHandler(
    enable_thinking=False,
    clip_model_path="path/to/mmproj-f16.gguf",
    use_gpu=True,       # <-- triggers the bug
    verbose=False,
)

llm = Llama(
    model_path="path/to/model.Q6_K.gguf",
    chat_format=CHAT_FORMAT_NAME,
    chat_handler=chat_handler,
    n_ctx=32768,
)

messages = [{
    "role": "user",
    "content": [
        {"type": "image_url", "image_url": {"url": "data:image/jpeg;base64,..."}},
        {"type": "text", "text": "Describe this image."},
    ],
}]
llm.create_chat_completion(messages=messages)
```

## Error Output

```
llama_context: n_ctx_seq (32768) < n_ctx_train (131072) -- the full capacity of the model will not be utilized
llama_kv_cache_iswa: using full-size SWA cache (ref: https://github.com/ggml-org/llama.cpp/pull/13194)
init_audio: audio input is in experimental stage and may have reduced quality:
    https://github.com/ggml-org/llama.cpp/discussions/13759

-- then immediately crashes with --

Assertion failed!
Program: ...\llama_cpp\lib\llama.dll
File: ...\src\llama-sampler.cpp
Line: 1094
Expression: found

For information on how your program can cause an assertion
failure, see the Visual C++ documentation on asserts
(Press Retry to debug the application -- JIT must be enabled)
```

## Root Cause Analysis

In `src/llama-sampler.cpp:1078-1094`, `llama_sampler_dist` performs softmax + weighted random sampling:

```c
bool found = false;
for (size_t i = 0; i < cur_p->size; ++i) {
    if (!found) {
        sum_run += cur_p->data[i].p;
        if (sum_run >= sum_tgt) {
            cur_p->selected = i;
            found = true;
        }
    }
    cur_p->data[i].p /= sum_cum;
}

// fallback to the last token (don't think this can happen)
assert(found);  // line 1094 -- fires with gemma4uv + GPU
```

When `gemma4uv` projector runs on GPU, the vision embeddings cause logits to become NaN or all `-inf`. After softmax, `sum_cum` is 0 and all token probabilities are 0, so no token can be selected -> `found` stays `false` -> assertion fires.

## Workaround

Set `use_gpu=False` on the mtmd handler. The CLIP/vision encoder runs on CPU, but the LLM backbone still uses GPU layers normally. Text-only inference is unaffected.

## Additional Notes

- mmproj loads successfully with `use_gpu=True`: `clip_model_loader` prints all 11 tensors correctly, vision + audio encoders detected
- Does **NOT** reproduce with `use_gpu=False` -- vision inference completes normally
- `gemma4uv` projector is relatively new -- likely a precision issue in the GPU CLIP path producing degenerate floating-point values

---
*This issue was drafted and submitted by an AI assistant (Claude) on behalf of the user, who encountered this bug during local testing.*
